### DISCUSSION TITLE
Write JSON with changing file name

### DISCUSSION THREAD

**Discussion by Seth-Julien-de-Lampon** [kedro-org/kedro#1659] [Idea]
For a use-case we are creating we like to write the final predictions of our models (alerts) as a JSON, to be picked up by the consumer.
For example, running the pipeline each 30 min. should create the following files:
-  results/alerts_2022_07_01_13_30.json
-  results/alerts_2022_07_01_14_00.json
-  results/alerts_2022_07_01_14_30.json
-  ....

I understand that the idea behind kedro is reproducibility, but for this use-case we want each run to write away a file with a different file name (sometimes even multiple files, all with unique names per run). 

Having the file_path set in data catalog, how would someone go about this?

**Comment by datajoely** [comment]
So I got halfway through writing how you could define a simple custom dataset to do the format you suggest here... but if you just set `versioned: True` in your catalog I think you may get the desired state with a slightly different date format?

**Reply by Seth-Julien-de-Lampon** [reply]
Exactly what I'm looking for, not sure why I overlooked this functionality. Thanks.

**Reply by datajoely** [reply]
This is my favourite kind of user problem! Have a great day and shout if you have any other problems 😀
